### ISSUE TITLE
[Accessibility][UX] Mobile: .menu-toggle-button should have a focus halo

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3214]
...follow-up to #3195

### Steps to reproduce (if reporting a bug)
- Open page and switch to mobile (either using browser dev tools, or by making the browser window narrow enough)
- Start tabbing

...notice the focus halo that items receive as you are tabbing. Pause for a moment when you have reached the "Login"/"Log out" menu item. Tabbing once more does not give any indication for it, but you have in fact already tabbed to the menu toggle. Hit space a couple of times, and you will see the mobile menu being expanded collapsed.

### Actual behavior (if reporting a bug)
There should be indication that the menu toggle is actually focused.

**Comment by stpaultim** [comment]
This looks like another good issue for OpenForce2022. Please, let us know if you have any questions?

**Comment by RakeshPotnuru** [comment]
Issue opened 4 years ago, is this still not resolved?

**Comment by stpaultim** [comment]
@RakeshPotnuru - We have over 2000 open issues in the issue queue, many of them are several years old, some older than this one.

The best thing to do in this situation - is to test the problem and see if is still a problem. If you can not recreate the problem, you can post a note with your findings and close the issue (which is a really helpful contribution).

For this issue, @klonos gave pretty good steps to recreate. Please, test this to see if you can recreate the problem.

**Comment by stpaultim** [comment]
I see now that someone tried to recreate this issue and failed. I'll see what I can do to see if this is still a problem unless @klonos is around and can help before me.

**Comment by stpaultim** [comment]
@RakeshPotnuru - I am able to recreate this one, you can test this with any site even a demo site. https://backdropcms.org/demo

1) Open the site and make it narrow enough that the hamburger menu is exposed.
2) Start to tab through the site. 
![image](https://user-images.githubusercontent.com/3144571/157600453-24875e83-2424-4318-8711-24713653e1da.png)

3) Hit tab again
![image](https://user-images.githubusercontent.com/3144571/157600620-73361a44-c581-41e0-8f6b-8fec53194a46.png)

4) Hit tab again and the hamburger menu should be active and should have the halo indicator, but it does not. 
![image](https://user-images.githubusercontent.com/3144571/157600787-d2378d74-f8f7-4805-9209-0f5d2b8b454b.png)

If you hit the space bar, you can confirm that the hamburger menu is active, because it will open and close. If you hit tab again, you will see the next item is selected.

**Comment by RakeshPotnuru** [comment]
I am working on this

**Comment by RakeshPotnuru** [comment]
I researched and found that `label` is not a focusable element.

**Comment by stpaultim** [comment]
@klonos or @jenlampton  - Do you have any ideas about how to solve this problem?

**Comment by jenlampton** [comment]
Can we try this? 

> Have the label element follow the input element in the HTML. Then use flex-direction: column-reverse to change its position to appear above the input element. You can then use the input:focus + label: {} to target the label.

From https://stackoverflow.com/questions/5978239/anyway-to-have-a-label-respond-to-focus-css#:~:text=Have%20the%20label%20element%20follow,%7B%7D%20to%20target%20the%20label.

**Comment by stpaultim** [comment]
@RakeshPotnuru - are you still working on this issue? Did you see the suggestion by @jenlampton? Do you have any questions?

**Comment by RakeshPotnuru** [comment]
@stpaultim 
I tried many solutions, not able to fix it.
Can you share the HTML file of this menu? Maybe we have to change the HTML a little bit.

**Comment by stpaultim** [comment]
@RakeshPotnuru - Thanks for trying. Feel free to work on other issues while you wait for feedback or additional ideas. 

@jenlampton, @klonos, or @indigoxela - any ideas for @RakeshPotnuru?

**Comment by klonos** [comment]
@RakeshPotnuru @stpaultim as @jenlampton pointed out, there may be a possible solution/workaround, but it would mean that we'd need to change the way the smartmenus library outputs code. This means that we have the following options:
- "hack" the 3rd party library to output HTML in a way that will allow us to implement the workaround
- wait for the maintainers of the 3rd-party library to provide a solution, and we then update the library Backdrop is using to a newer version that solves this problem.

Another option is that we provide a PR against the 3rd part library, to help speed things up.

**Comment by stpaultim** [comment]
@klonos - That makes is seem as if this is no longer a "good first issue." Am I correct?

**Comment by klonos** [comment]
I think so ...unfortunately 😞

**Comment by klonos** [comment]
@RakeshPotnuru if you still wanna have a go at this, the smartmenus library we are using is located under `/core/misc/smartmenus`.
